src/Microsoft.DotNet.Darc/DarcLib/GitHubClient.cs

```diff
@@ -435,11 +438,18 @@ public async Task MergeDependencyPullRequestAsync(string pullRequestUrl, MergePu
             MergeMethod = parameters.SquashMerge ? PullRequestMergeMethod.Squash : PullRequestMergeMethod.Merge
         };
 
-        await GetClient(owner, repo).PullRequest.Merge(owner, repo, id, mergePullRequest);
+        try
+        {
+            await gitHubClient.PullRequest.Merge(owner, repo, id, mergePullRequest);
+        }
+        catch (Octokit.PullRequestNotMergeableException notMergeableException)
+        {
+            throw new PullRequestNotMergeableException(notMergeableException.Message);
+        }
 
         if (parameters.DeleteSourceBranch)
         {
-            await GetClient(owner, repo).Git.Reference.Delete(owner, repo, $"heads/{pr.Head.Ref}");
+            await gitHubClient.Git.Reference.Delete(owner, repo, $"heads/{pr.Head.Ref}");
```

Comment:
We need to put `try/catch` around this possibly if the branch gets auto-deleted?

Reply:
Yes